### DISCUSSION TITLE
Filtering by exact relation ids

### DISCUSSION THREAD

**Discussion by redbaron76** [pocketbase/pocketbase#4443] [Q&A]
Hello, let's say I have a table like this `{
  "id": "nar1k1ddvq2wudg",
  "members": [
    "N7cPxIIx6ePY2YY",
    "sPLqLWTqs9SFqS9"
  ],
  "type": "DIRECT"
}`

Hot to get the `id` (nar1k1ddvq2wudg) filtering by `type` (DIRECT) and members that have to be exactly `["N7cPxIIx6ePY2YY","sPLqLWTqs9SFqS9"]`??

I've tried all the possible combinations using `.getFirstListItem(FILTER, ...)` but without success, thanks.

PS. `members` is a multiple relation field of "users"

**Comment by ganigeorgiev** [comment]
I'm not sure that I understand your question.

In general multivalued fields are stored as serialized json array, meaning that if you want an exact match of your array, you can compare it against its string value:

```js
await pb.collection("example").getFirstListItem(`members = '["N7cPxIIx6ePY2YY","sPLqLWTqs9SFqS9"]'`)
```

**Reply by redbaron76** [reply]
I need a filter to get the `id` of the first matched record which must be of type `DIRECT` and whose members are exactly ["N7cPxIIx6ePY2YY","sPLqLWTqs9SFqS9"] (or at least ["sPLqLWTqs9SFqS9","N7cPxIIx6ePY2YY"])...

**Reply by ganigeorgiev** [reply]
Then in this case something like this should work:

```js
await pb.collection("example").getFirstListItem(`type="DIRECT" && (members='["N7cPxIIx6ePY2YY","sPLqLWTqs9SFqS9"]' || members='["sPLqLWTqs9SFqS9","N7cPxIIx6ePY2YY"]')`)
```

**Reply by redbaron76** [reply]
It works without parenthesis around the `(members = [...] || members = [...])`.
Thanks anyway!
